### PR TITLE
Issue: #214481 Add Option to Ignore Code Blocks in Text-to-Speech

### LINKED ISSUE #214481: Text to speech might not vocalizes on code fences
The text-to-speech function vocalizes the content within **code fences** in Copilot's responses, which typically contain code and are not designed for vocalization. Could there be a feature to bypass the code fences in text-to-speech?

### PR DESCRIPTION
**Pull request by jaymroy**
fix #214481

Implementation of a requested feature
I introduced a new setting under the existing accessibility.voice.* namespace, modified the nextChatResponseChunk generator to track code block boundaries and stopped synthesizing text within the code blocks when the setting is enabled.

### CODE REVIEW

**Comment by HarshPatel2010** [comment]
Preparing review...

**Comment by HarshPatel2010** [comment]
Preparing review...

**Comment by HarshPatel2010** [comment]
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 3 🔵🔵🔵⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<a href='https://github.com/microsoft/vscode/pull/235697/files#diff-3fd6fda06fcf4ddc95190c5bd20dfff5058b91c68c8ab09ed5a97c2d90b60db8R771-R795'><strong>Code Complexity</strong></a><br>The implementation of the feature to ignore code blocks in text-to-speech adds significant complexity to the `nextChatResponseChunk` method. Consider refactoring for better maintainability.

<a href='https://github.com/microsoft/vscode/pull/235697/files#diff-9d33df30c0953d8c43bdb5285db467b65a408ea054206635f08ef026f7ccbe27R141-R141'><strong>Typo in Constant</strong></a><br>There is a typo in the constant name 'IgnoreCodeBlocks' which is spelled as 'accesibility.voice.ignoreCodeBlocks' instead of 'accessibility.voice.ignoreCodeBlocks'.

</td></tr>
</table>

**Review by bpasero — CHANGES_REQUESTED** [review]
The approach in principle makes sense to me but it does not work: the chunk from here:

```
const { chunk, offset } = this.parseNextChatResponseChunk(response, totalOffset);
```

No longer contains any backticks. They are removed when calling `renderStringAsPlaintext` I believe, can you check?

**Review by bpasero — DISMISSED** [review]

**Comment by bpasero** [comment]
Thanks!

**Review by dbaeumer — DISMISSED** [review]

**Comment by bpasero** [comment]
@jaymroy can you agree to the CLA bot?

**Comment by jaymroy** [comment]
> License

@microsoft-github-policy-service agree

**Review by meganrogge — APPROVED** [review]

File: src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts [hunk 1]
```diff
@@ -707,6 +707,11 @@ class ChatSynthesizerSessionController {
 	}
 }
 
+interface IChatSyntheSizerContext {
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts:710]: Shouldn't `SyntheSizer` be `Synthesizer`?

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**